### PR TITLE
Change update method for `boot_disk.auto_delete` on `google_compute_instance`

### PR DESCRIPTION
**Pull request by karolgorc**
closes https://github.com/hashicorp/terraform-provider-google/issues/14707
closes https://github.com/hashicorp/terraform-provider-google/issues/6892

This will prevent recreating the VM when changing `auto_delete` parameter

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: `google_compute_instance` is no longer recreated when changing `boot_disk.auto_delete`
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Hi @karolgorc,
I've found that you have many open pull requests for Magic Modules, and wanted to reach you about the upcoming language migration. I've randomly picked this pull request to comment on.

You can view the details of the language migration here: https://github.com/hashicorp/terraform-provider-google/issues/19583

Essentially, later this week many of your pull requests may become incompatible due to the template files changing from Ruby (.erb) to Go (.tmpl) templates. Our team (Magic Modules repository maintainers) has some tooling to automatically convert changes to the new language, and we can prepare a converted commit for all of your open pull requests.

However, this converted commit will be rebased to after the migration. This means in order to push the new changes to your pull request, we would need to **force push** the commit to your fork's branch. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event anything went wrong or needs additional changes.

Of course, you also have the option to convert your pull requests yourself after the migration tomorrow. We have links to a preview branch and updated documentation in the link above.

As an owner of many open pull requests, what would you like to do? Do we have your permission to push commits to your fork's branches, or would you like to convert them yourself?

Let me know if you have any questions or if there is a better method to converse.

**Comment by c2thorn** [comment]
@karolgorc please see this comment: https://github.com/GoogleCloudPlatform/magic-modules/pull/11742#issuecomment-2372005694

If you would like assistance converting files, we need your permission to push changes to this branch.

**Comment by karolgorc** [comment]
Hi @c2thorn.

Thanks for the info. I see that the changes are already live. The branches affected are
- auto-delete-update
- resource-policies-diff
- key-revocation-action-field
- min-cpu-platform-error-message
- remove-hardcoded-values-instance-template
- custom-machine-type-doc-update

You can force-push to the fork. Do you need me to do something in order for this to work?

**Comment by c2thorn** [comment]
> Hi @c2thorn.
> 
> Thanks for the info. I see that the changes are already live. The branches affected are
> 
> * auto-delete-update
> * resource-policies-diff
> * key-revocation-action-field
> * min-cpu-platform-error-message
> * remove-hardcoded-values-instance-template
> 
> You can force-push to the fork. Do you need me to do something in order for this to work?

Thanks @karolgorc!
If things go smoothly, you won't have to do anything. But to help in the case we need to restore your branch from our backups, could you add me (c2thorn) as collaborator to your fork here:
https://github.com/karolgorc/magic-modules/settings

**Comment by karolgorc** [comment]
done

**Comment by KevsAlex** [comment]
is this still in review or was it solved in some version ?

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl [hunk 1]
```diff
@@ -2411,6 +2410,19 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
...
+	if d.HasChange("boot_disk.0.auto_delete") {
+		op, err := config.NewComputeClient(userAgent).Instances.SetDiskAutoDelete(project, zone, instance.Name, d.Get("boot_disk.0.auto_delete").(bool), d.Get("boot_disk.0.device_name").(string)).Do()
+		if err != nil {
+			return errwrap.Wrapf("Error setting auto_delete : {{"{{"}}err{{"}}"}}", err)
+		}
+		opErr := ComputeOperationWaitTime(config, op, project, "changing auto_delete", userAgent, d.Timeout(schema.TimeoutUpdate))
+		if opErr != nil {
+			return opErr
+		}
+		log.Printf("[DEBUG] Successfully changed auto_delete on disk %s", instance.Disks[0].DeviceName)
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl:2423]:
```suggestion
```

**Comment by nightpool** [comment]
Would love to see this get merged!

**Comment by karolgorc** [comment]
Hi can this get reviewed or reassigned?

**Comment by rileykarson** [comment]
Reassigning to @slevenick

Also /gcbrun

**Review by slevenick — APPROVED** [review]

**Pull request merged**